### DISCUSSION TITLE
Adaptive streaming with multiple codecs using HLS and DASH: Any guides?

### DISCUSSION THREAD

**Discussion by Stefan-Olt** [videojs/video.js#7951] [General]
Hi,
I try to have the following functionality with video.js, unfortunately I cannot get it to work, I'm not sure if that's even possible at all and what plug-ins are needed: I would like to have video.js bandwidth/quality/codec adapting to the user system/connection. I figured out the only way to do that is HLS and DASH (correct me if I'm wrong).

So my idea is to offer the following formats in this priority:
DASH: Video: AV1, VP9, H.264 (high profile); Audio: Opus, AAC
HLS (using MPEG2-TS): Video: H.264 (high profile); Audio: AAC (HLS could be first choice for Apple devices)
Non-adaptive fallback:
H.264 (main profile)+AAC / mp4
VP8+Vorbis / webm

While HLS works fine, I cannot get DASH working. I'm not even sure I encode it correctly (I use ffmpeg), as there are a lot of options and the entire concept seems to me a lot more complicated than HLS (I verified the stream files have been requested from the server). I read that the integrated VHS-plugin also supports DASH, but somewhere was written only MPEG, not WebM, but ffmpeg generates m4s files for AV1 by default... then there is another DASH plug-in...
Is there any up-to-date guide that shows what's possible and what's not, what plug-ins to use and maybe example files to test and/or a ffmpeg command line that will create compatible files.

Best regards
Stefan

**Comment by cezarfloroiu** [comment]
Have you had any luck with this?  Trying to play a live stream that uses AV1 codec (via cloudflare) and I couldnt make the DASH working. And docs seem really weak in this regards. Thanks!
